### PR TITLE
Converting to portable app

### PR DESCRIPTION
**Pull request by JunTaoLuo**
This needs https://github.com/aspnet/Hosting/pull/719

cc @Tratcher @pakrym

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
:up::date:

File: ServerTests.sln [hunk 1]
```diff
@@ -16,6 +16,16 @@ Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "ServerComparison.Functional
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "ServerComparison.TestSites", "test\ServerComparison.TestSites\ServerComparison.TestSites.xproj", "{030225D8-4EE8-47E5-B692-2A96B3B51A38}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNetCore.Server.Testing", "..\Hosting\src\Microsoft.AspNetCore.Server.Testing\Microsoft.AspNetCore.Server.Testing.xproj", "{3DA89347-6731-4366-80C4-548F24E8607B}"
```
Comment (Tratcher) [inline ServerTests.sln]: Revert

File: test/ServerComparison.FunctionalTests/Helpers.cs [hunk 2]
```diff
@@ -8,9 +8,9 @@ namespace ServerComparison.FunctionalTests
 {
     public class Helpers
     {
-        public static string GetApplicationPath()
+        public static string GetApplicationPath(ApplicationType applicationType)
         {
-            return Path.GetFullPath(Path.Combine("..", "..", "..", "..", "..", "ServerComparison.TestSites"));
+            return Path.GetFullPath(Path.Combine("..", "..", "..", "..", "..", applicationType == ApplicationType.Standalone? "ServerComparison.TestSites.Standalone" : "ServerComparison.TestSites"));
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/Helpers.cs]: line wrap

File: test/ServerComparison.FunctionalTests/ResponseTests.cs [hunk 3]
```diff
@@ -16,133 +16,142 @@
...
+            return ResponseFormats(serverType, runtimeFlavor, architecture, applicationBaseUrl, CheckManuallyChunkedAndCloseAsync, applicationType);
         }
 
-        public async Task ResponseFormats(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl, Func<HttpClient, ILogger, Task> scenario)
+        public async Task ResponseFormats(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl, Func<HttpClient, ILogger, Task> scenario, ApplicationType applicationType)
         {
             var logger = new LoggerFactory()
                             .AddConsole()
-                            .CreateLogger(string.Format("ResponseFormats:{0}:{1}:{2}", serverType, runtimeFlavor, architecture));
+                            .CreateLogger(string.Format("ResponseFormats:{0}:{1}:{2}", serverType, runtimeFlavor, architecture, applicationType));
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseTests.cs]: You need to update the format string

File: test/ServerComparison.TestSites/web.config [hunk 4]
```diff
@@ -4,6 +4,6 @@
     <handlers>
       <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
     </handlers>
-    <aspNetCore processPath=".\ServerComparison.TestSites.exe" arguments="" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="true" />
+    <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="false"/>
```
Comment (Tratcher) [inline test/ServerComparison.TestSites/web.config]: forwardWindowsAuthToken="true"

File: test/ServerComparison.FunctionalTests/NtlmAuthenticationTest.cs [hunk 5]
```diff
@@ -21,24 +21,26 @@ public class NtlmAuthenticationTests
...
+        // [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:5050/", ApplicationType.Portable)]
+        // [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:5051/", ApplicationType.Portable)]
+        // [InlineData(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:5052/", ApplicationType.Portable)]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:5052/", ApplicationType.Portable)]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:5053/", ApplicationType.Standalone)]
+        public async Task NtlmAuthentication(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl, ApplicationType applicationType)
         {
             var logger = new LoggerFactory()
                             .AddConsole()
                             .CreateLogger(string.Format("Ntlm:{0}:{1}:{2}", serverType, runtimeFlavor, architecture));
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/NtlmAuthenticationTest.cs]: Log the app type

**Comment by Tratcher** [comment]
:shipit:

File: test/ServerComparison.FunctionalTests/HelloWorldTest.cs [hunk 6]
```diff
@@ -20,59 +20,63 @@ public class HelloWorldTests
...
 
         [ConditionalTheory]
         [SkipIfIISVariationsNotEnabled]
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
         [SkipIfCurrentRuntimeIsCoreClr]
-        [InlineData(ServerType.IIS, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:5069/")]
-        //[InlineData(ServerType.IIS, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:5070/")]
-        public Task HelloWorld_IIS_X86(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        [InlineData(ServerType.IIS, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:5072/", ApplicationType.Portable)]
```
Comment (BrennanConroy) [inline test/ServerComparison.FunctionalTests/HelloWorldTest.cs]: x64 on HelloWorld_IIS_X86 :/ I know this wasn't you but can you fix it?

File: test/ServerComparison.FunctionalTests/HelloWorldTest.cs [hunk 7]
```diff
@@ -20,59 +20,63 @@ public class HelloWorldTests
...
+            return HelloWorld(serverType, runtimeFlavor, architecture, applicationBaseUrl, applicationType);
         }
 
         [Theory]
-        //[InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:5065/")]
-        [InlineData(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:5066/")]
-        [InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:5067/")]
-        public Task HelloWorld_Kestrel(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        //[InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:5066/", ApplicationType.Portable)]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:5067/", ApplicationType.Portable)]
```
Comment (kichalla) [inline test/ServerComparison.FunctionalTests/HelloWorldTest.cs]: intentional? its currently Portable, should it be Standalone?

Comment (JunTaoLuo) [inline test/ServerComparison.FunctionalTests/HelloWorldTest.cs]: Application type is ignored by the deployers for CLR apps. However, it is used to pick the correct project. We are testing portable and standalone for CoreCLR apps separately though.

**Pull request merged**